package/skylark_download_daemon/src/skylark_download_daemon.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2017 Swift Navigation Inc.
+ * Contact: Mark Fine <mark@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <libskylark.h>
+
+#define VERBOSE
+
+//
+// Download Daemon - connects to Skylark and receives SBP messages.
+//
+
+static size_t download_callback(void *p, size_t size, size_t n, void *up)
+{
+  int *fd = (int *)up;
+  ssize_t m = write(*fd, p, size * n);
+  return m;
+}
```

Comment:
This callback is super generic and I think it can be left in libskylark and doesn't need to be passed in as an argument.